plugin/trino-hive/src/main/java/io/trino/plugin/hive/metastore/file/FileHiveMetastore.java

```diff
@@ -532,7 +533,17 @@ private List<TableInfo> listAllTables(String databaseName)
         return listTablesCache.getUnchecked(databaseName);
     }
 
-    private synchronized List<TableInfo> doListAllTables(String databaseName)
+    @Override
+    public synchronized List<String> getTablesWithParameter(String databaseName, String parameterKey, Set<String> parameterValues)
+    {
+        requireNonNull(parameterKey, "parameterKey is null");
+        requireNonNull(parameterValues, "parameterValue is null");
+        return doListAllTables(databaseName, table -> parameterValues.contains(table.getParameters().get(parameterKey))).stream()
```

Comment:
Do we need to handle the case where `table.getParameters().get(parameterKey)` is null separately ? That potentially is nullptr exception

Reply:
I'm a bit confused about how changing the type helps.
I was thinking about the case where given parameter key does not exist in table parameters. I think the safe thing to do would be to return empty list of table names in that case.

Reply:
A `Set` can contain `null` values so theoretically an API with `Set` could mean a caller would like to filter for null or non-existent values. `ImmutableSet` cannot contain null, and additionally, it is safe to check for null existence there. This means that ImmutableSet better describes the API here which only supports non-null values for the filter

Reply:
That seems a subtle distinction, it would be easy for users of the API to miss it.
Can we just handle that corner case safely instead ?

Reply:
How can they miss it? With `ImmutableSet` there is no way to pass the wrong argument

Reply:
We can't pass in a key for which there is no entry in table parameters ?

Reply:
They can, that is actually the point of this to return only those that have it. But that works ok becasue the parameters values cannot contain null thus potential mistake to try to filter by null value will just not work and it is safe to check for null element becasue `ImmutableSet` as all immutable collections explicitly allows for null argument to `com.google.common.collect.ImmutableCollection#contains`.

`public abstract boolean contains(@CheckForNull Object object);`